emails/build-emails

```diff
@@ -28,17 +36,41 @@ function renderTemplate(createTemplate, name) {
     templateText = handlebars.compile(templateText)(mockData)
   }
 
-  fs.mkdirSync(DIST_DIR, { recursive: true })
-  fs.writeFileSync(path.join(DIST_DIR, name + '.html'), template, 'utf8')
-  fs.writeFileSync(path.join(DIST_DIR, name + '.txt'), templateText, 'utf8')
+  return {subject, name, template, templateText}
+}
+
+async function syncPostmark(files) {
+  if (!POSTMARK_SERVER_API_TOKEN) {
+    console.error('You must specify a valid POSTMARK_SERVER_API_TOKEN environment variable.')
+    process.exit(1)
+  }
+  const client = new postmark.ServerClient(POSTMARK_SERVER_API_TOKEN)
+  for (const filename of files) {
+    const { subject, name, template, templateText } = renderTemplate(filename)
+    const params = {
+      Name: name,
+      Alias: name,
+      Subject: subject,
+      HtmlBody: template, 
+      TextBody: templateText
+    }
+    try {
+      await client.getTemplate(name)
+      await client.editTemplate(name, params)
+    } catch {
+      await client.createTemplate(params)
+    }
+  }
 }
 
 function generateTemplates(files) {
   execSync(`rm -rf "${DIST_DIR}"`)
 
   files.forEach(filename => {
-    const template = require(path.join(__dirname, 'templates', filename))
-    renderTemplate(template, path.basename(filename, '.js'))
+    const { name, template, templateText } = renderTemplate(filename)
+    fs.mkdirSync(DIST_DIR, { recursive: true })
```

Comment:
Should we lift this to be outside the for loop?

Reply:
Indeed, it was in the loop previously